app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/MeetingsFragmentTest.kt

```diff
@@ -0,0 +1,326 @@
+import com.github.sdp_begreen.begreen.services.GeocodingService
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runTest
+import org.hamcrest.Matchers.equalTo
+import org.hamcrest.Matchers.`is`
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.koin.dsl.module
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import java.io.IOException
+import java.text.DateFormat
+import java.util.Calendar
+import java.util.Locale
+
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class MeetingsFragmentTest {
+
+    companion object {
+        private val meetingService: MeetingService = mock(MeetingService::class.java)
+        private val participantService: MeetingParticipantService =
+            mock(MeetingParticipantService::class.java)
+        private val geocoderApi: GeocodingService = mock(GeocodingService::class.java)
+        private val db: DB = mock(DB::class.java)
+        private val auth: Auth = mock(Auth::class.java)
+
+        private val initiallyConnectedUser = User("123456", 10, "User 1")
+
+        private val meetings = listOf(
+            Meeting(
+                "m1",
+                "John",
+                "Downtown Cleanup",
+                "Let's meet up and clean up the downtown area",
+                1718445600000,
+                1718467200000,
+                CustomLatLng(37.7749, -122.4194),
+                CustomLatLng(37.7749, -122.4194)
+            ),
+            Meeting(
+                "m2",
+                "Sarah",
+                "Park Cleanup",
+                "Let's meet up and clean up the local park",
+                1718186400000,
+                1718208000000,
+                CustomLatLng(37.7739, -122.4312),
+                CustomLatLng(37.7739, -122.4312)
+            ),
+            Meeting(
+                "m3",
+                "Tom",
+                "Beach Cleanup",
+                "Let's meet up and clean up the local beach",
+                1710237600000,
+                1710252000000,
+                CustomLatLng(46.517355, 6.628854),
+                CustomLatLng(46.517355, 6.628854)
+            ),
+            Meeting(
+                "m4",
+                "Andree",
+                "School Cleanup",
+                "Let's meet up and clean up the local school",
+                1710237600000,
+                1710238600000,
+                CustomLatLng(0.0, 0.0),
+                CustomLatLng(0.0, 0.0),
+            )
+        )
+
+        private val meetingsFlow = MutableStateFlow(meetings)
+        private val localities = listOf("Paris", "Berlin", "London", "")
+
+        val addRemoveParticipantChannel = Channel<String>(1)
```

Comment:
Yes I totally agree with you. I wan't sure about it either, but couldn't come with a better one.

It is supposed to represent a `Channel` where both `add participant` and `remove participant` can be send through. So that I only declare one `Channel` and can use it in multiple test.

Do you have any idea for a better name ?

Reply:
I will add this thanks ^^